src/utils/mutex.js

```diff
@@ -0,0 +1,68 @@
+'use strict'
+
+const assert = require('assert')
+const mortice = require('mortice')
+const nanoid = require('nanoid')
+const setImmediate = require('async/setImmediate')
+const noop = () => {}
+
+// Wrap mortice to present a callback interface
+class Mutex {
+  constructor (repoOwner, options = {}) {
+    // Ensure that we get a different mutex for each instance of the lock
+    const randId = nanoid()
+    this.mutex = mortice(randId, {
+      singleProcess: repoOwner
+    })
```

Comment:
we need to handle the `global` option
either we do this here or inside mortice, imo the second option is better.

i have added globalThis support to ipfs-utils here https://github.com/ipfs/js-ipfs-utils/blob/master/src/globalthis.js

`const globalThis = require('ipfs-utils/src/globalthis')`
or even just 
`require('ipfs-utils/src/globalthis')`

@achingbrain thoughts ? should we PR mortice ?

Reply:
> we need to handle the `global` option

I'm not sure I follow - what's the problem you are trying to solve?

Reply:
mortice defaults the global option to `global` which isnt safe for all js envs

Reply:
with my suggestion you can remove the option and it will always have the right globalThis

Reply:
Gotcha. Has this caused problems?

I've put a PR together that removes the option [here](https://github.com/achingbrain/mortice/pull/2).

Reply:
I ask because mortice has tests that runs across browsers, webworkers, node and node cluster and they all pass, so if it's detecting the wrong global as it stands, I'd like to know about it!

Reply:
you assume browserify is used and global is defined not always true